services/galley/test/integration/API/Federation.hs

```diff
@@ -0,0 +1,75 @@
+-- This file is part of the Wire Server implementation.
+--
+-- Copyright (C) 2020 Wire Swiss GmbH <opensource@wire.com>
+--
+-- This program is free software: you can redistribute it and/or modify it under
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+
+module API.Federation where
+
+import API.Util
+import Bilge
+import Bilge.Assert
+import Control.Lens
+import Data.List1
+import Galley.Types
+import Imports
+import Test.Tasty
+import Test.Tasty.HUnit
+import TestHelpers
+import TestSetup
+import Wire.API.Federation.API.Galley (GetConversationsRequest (..))
+import qualified Wire.API.Federation.API.Galley as FedGalley
+
+tests :: IO TestSetup -> TestTree
+tests s =
+  testGroup
+    "federation"
+    [ test s "getConversations (/federation/conversations/get-by-ids) : All Found" getConversationsAllFound
+    ]
+
+getConversationsAllFound :: TestM ()
+getConversationsAllFound = do
+  [alice, bob] <- randomUsers 2
+  connectUsers alice (singleton bob)
+  -- create & get one2one conv
+  cnv1 <- responseJsonUnsafeWithMsg "conversation" <$> postO2OConv alice bob (Just "gossip1")
+  getConvs alice (Just $ Left [cnvId cnv1]) Nothing !!! do
+    const 200 === statusCode
+    const (Just [cnvId cnv1]) === fmap (map cnvId . convList) . responseJsonUnsafe
+  -- create & get group conv
+  carl <- randomUser
+  connectUsers alice (singleton carl)
+  cnv2 <- responseJsonUnsafeWithMsg "conversation" <$> postConv alice [bob, carl] (Just "gossip2") [] Nothing Nothing
+  getConvs alice (Just $ Left [cnvId cnv2]) Nothing !!! do
+    const 200 === statusCode
+    const (Just [cnvId cnv2]) === fmap (map cnvId . convList) . responseJsonUnsafe
+  -- get both
+
+  fedGalleyClient <- view tsFedGalleyClient
+  cs <- Just <$> FedGalley.getConversations fedGalleyClient (GetConversationsRequest alice [cnvId cnv1, cnvId cnv2])
```

Comment:
If you replace `alice` in this line with another random user not part of that conversation, I suppose this test will fail, right? getConversations should then return a non-200 status code or some other embedded error? Could you add a test for the failure case of the requesting user not being part of the conversation?

I think we need to add some FUTUREWORK here to replace alice with a remote user once we have support for that.

Reply:
Good point. I changed the behaviour and added a test: If the user is not part of the conversation it is not included in the result. Rationale: We want to handle this case, because it might result frequent when two backends are not fully synced. This requires lifiting the problem to the value level. Excluding the conversation from the result is also consistent with the case when a conversation id cannot be found.